src/Collection/CollectionTrait.php

```diff
@@ -702,7 +702,7 @@ public function isEmpty()
     public function unwrap()
     {
         $iterator = $this;
-        while (get_class($iterator) === 'Cake\Collection\Collection') {
+        while ($iterator instanceof Collection) {
```

Comment:
I think this was checking for the specific class as there are a few subclasses that have to be treated differently.